glibc 2.17 install fails when compiling on Ubuntu 24.04

Makeinfo fails due do bad syntax of the input file (fixed in later glibc versions).
That results in libc.info file not being generated.
The install target then does
for file in libc.info*; ...
Which in the past seems to silently have done nothing if libc.info does not exist, but these days tries to install a non-existing file called "libc.info*".
I tried forcing it to use BASH as shell here but I had no success.
The simplest solution I found was to properly prevent it from creating libc.info by adding
echo "ac_cv_prog_MAKEINFO=" >>config.cache
to scripts/build/libc/glibc.sh

I encountered the same issue. The log is as follows:
`
[ALL  ]      mv -f /home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/dir-add.texi.new /home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/dir-add.texi
[ALL  ]      LANGUAGE=C LC_ALL=C makeinfo -P /home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/ --output=/home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/libc.info libc.texinfo
[ALL  ]      texi2any (XS parser): bug: parent hv not already set
[ALL  ]      
[ALL  ]      /home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/buildtools/bin/makeinfo: line 2: 897972 Aborted                 (core dumped) /usr/bin/makeinfo --force "${@}"
[ALL  ]      .././scripts/mkinstalldirs /home/u/crosstool-ng-glibc-2.27/x86_64-ubuntu16.04-linux-gnu/x86_64-ubuntu16.04-linux-gnu/sysroot/usr/share/info
[ALL  ]      mkdir -p -- /home/u/crosstool-ng-glibc-2.27/x86_64-ubuntu16.04-linux-gnu/x86_64-ubuntu16.04-linux-gnu/sysroot/usr/share/info
[ALL  ]      for file in /home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/libc.info*; do   /home/u/crosstool-ng-glibc-2.27/.build/tools/bin/install -c -m 644 $file /home/u/crosstool-ng-glibc-2.27/x86_64-ubuntu16.04-linux-gnu/x86_64-ubuntu16.04-linux-gnu/sysroot/usr/share/info/; done
[ALL  ]      /usr/bin/install: cannot stat '/home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib/manual/libc.info*': No such file or directory
[ERROR]      make[4]: *** [Makefile:190: /home/u/crosstool-ng-glibc-2.27/x86_64-ubuntu16.04-linux-gnu/x86_64-ubuntu16.04-linux-gnu/sysroot/usr/share/info/libc.info] Error 1
[ALL  ]      make[4]: Leaving directory '/home/u/crosstool-ng-glibc-2.27/.build/src/glibc-2.23/manual'
[ERROR]      make[3]: *** [Makefile:214: manual/subdir_install] Error 2
[ALL  ]      make[3]: Leaving directory '/home/u/crosstool-ng-glibc-2.27/.build/src/glibc-2.23'
[ERROR]      make[2]: *** [Makefile:12: install] Error 2
[ALL  ]      make[2]: Leaving directory '/home/u/crosstool-ng-glibc-2.27/.build/x86_64-ubuntu16.04-linux-gnu/build/build-libc/multilib'
[ERROR]  
[ERROR]  >>
[ERROR]  >>  Build failed in step 'Building for multilib 1/1: '''
[ERROR]  >>        called in step 'Installing C library'
[ERROR]  >>        called in step '(top-level)'
[ERROR]  >>
[ERROR]  >>  Error happened in: CT_DoExecLog[scripts/functions@378]
[ERROR]  >>        called from: glibc_backend_once[scripts/build/libc/glibc.sh@273]
[ERROR]  >>        called from: CT_IterateMultilibs[scripts/functions@1610]
[ERROR]  >>        called from: glibc_main[scripts/build/libc/glibc.sh@26]
[ERROR]  >>        called from: do_libc_main[scripts/build/libc.sh@33]
[ERROR]  >>        called from: main[scripts/crosstool-NG.sh@712]
[ERROR]  >>
[ERROR]  >>  For more info on this error, look at the file: 'build.log'
[ERROR]  >>  There is a list of known issues, some with workarounds, in:
[ERROR]  >>      https://crosstool-ng.github.io/docs/known-issues/
[ERROR]  >>
[ERROR]  >>  If you feel this is a bug in crosstool-NG, report it at:
[ERROR]  >>      https://github.com/crosstool-ng/crosstool-ng/issues/
[ERROR]  >>
[ERROR]  >>  Make sure your report includes all the information pertinent to this issue.
[ERROR]  >>  Read the bug reporting guidelines here:
[ERROR]  >>      http://crosstool-ng.github.io/support/
[ERROR]  
[ERROR]  (elapsed: 43:41.13)
`

--- a/config/libc/glibc.in
+++ b/config/libc/glibc.in
@@ -289,6 +289,12 @@ comment "|  Building GLIBC locales requires that GLIBC supports        "
 comment "|  the build machine as the target.                           "
 endif # LIBC_GLIBC && GLIBC_LOCALES
 
+# Older Glibc versions are incompatible with newer makeinfo versions. Skip
+# building the user manual if needed.
+config GLIBC_MAKEINFO_WORKAROUND
+    def_bool y
+    depends on GLIBC_2_23_or_older
+
 if KERNEL_LINUX
 
 choice GLIBC_SUPPORTED_KERNEL

--- a/scripts/build/libc/glibc.sh
+++ b/scripts/build/libc/glibc.sh
@@ -182,6 +182,10 @@ glibc_backend_once()
     # or even after they get installed...
     echo "ac_cv_path_BASH_SHELL=/bin/bash" >>config.cache
 
+    if [ "${CT_GLIBC_MAKEINFO_WORKAROUND}" = "y" ]; then
+        echo "ac_cv_prog_MAKEINFO=" >>config.cache
+    fi
+
     CT_SymlinkToolsMultilib
 
     # Configure with --prefix the way we want it on the target...

